### PR TITLE
so github users can become more familiar with bash

### PR DESCRIPTION
**Pull request by gh0st**
This is just kind of a draft. I can add more material or make comments more descriptive once it's been placed inside a PDF. This is for #399

### CODE REVIEW

**Comment by gh0st** [comment]
Not sure why the Travis build is failing...

**Comment by beardofedu** [comment]
@gh0st 
First, thanks for creating this! 

Try building the content locally, you can run `script/server` from your Terminal to build the page. It might have something to do with the `{& capture}` content.

Additionally, before you start the actual content of the page, throw the following above it:

```
main-content: | 

```

Then tab all of the content that is considered the 'main content' (aka everything) so it is considered a child of the `main-content` identifier. If you continue to have issues, please @ mention me.

**Comment by gh0st** [comment]
Sorry @beardofedu, not sure how I'd run `script/server` from terminal to build the page.

**Comment by gh0st** [comment]
I take that back. I just get the following errors:
```
==> Installing gem dependencies…
script/bootstrap: line 8: bundle: command not found
script/bootstrap: line 9: bundle: command not found
ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
hostname: thereaper.local
pwd: /Users/chad/Developer/training-kit
git version 2.3.2 (Apple Git-55)
==> Installing gem dependencies…
script/bootstrap: line 8: bundle: command not found
script/bootstrap: line 9: bundle: command not found
./script/server: line 6: bundle: command not found
```

**Comment by hollenberry** [comment]
Hey @gh0st ! You may have just resolved this, as I see the comment is gone, but try running `script/setup` and let us know if that allows you to build 👍

**Comment by gh0st** [comment]
Having  an issue installing RedCloth 4.2.9 @beardofedu @hollenberry

**Comment by hollenberry** [comment]
@gh0st Can you possibly post a screenshot of the last few lines of that error?

**Comment by crichID** [comment]
@gh0st thank you for getting this started and creating such an awesome tool! I can't wait to have this available on the site. It looks like the cheatsheets are not building locally so I am working on that today!

> `main-content: |`

This is not actually required for the cheat sheets to render properly since they use a different layout. I fixed it in my most recent commit!

**Comment by gh0st** [comment]
Thank you @crichID!!!

**Comment by hectorsector** [comment]
👋 @crichID @gh0st the cheatsheet builds for me at http://localhost:4000/kit/downloads/github-bash-cheat-sheet.html ✨ 

screen shot 2017-01-05 at 6 36 39 pm

**Comment by gh0st** [comment]
Sweet!

**Comment by crichID** [comment]
@hectorsector I would guess it is building for you because you pulled the branch from @gh0st remote and built it from there. When you did, you may have noticed that the `script/server` built using submodules that were removed in https://github.com/github/training-kit/pull/396:

screenshot 2017-01-05 21 33 36

Building the french cheat sheet on master in @gh0st's remote results in:

screenshot 2017-01-05 21 29 49

However, if you switch back over to origin/master and try to access the same french cheat sheet, you get:

screenshot 2017-01-05 21 39 44

Incidentally, the same thing happens if you try to access the html version of the french cheat sheet on services.github.com:

screenshot 2017-01-05 21 42 43

But not if you access the Arabic or Chinese cheat sheets:

screenshot 2017-01-05 21 44 02

The main difference appears to be the `leadingpath:` in the yaml frontmatter for ar and zn_CN is `../../../` where fr is `../../`. Fixing it in the fr file on master showed this was correct on origin/master.

However, if you look at https://services.github.com/on-demand/downloads/github-git-cheat-sheet/, you will see that the english version of the html cheat sheet is also not rendering properly. Building origin/master locally shows the same result. The leading path for the english cheat sheet was `../`. I changed it to `../../` and it is not rendering correctly when I build on origin/master.

**tl;dr** I updated the leading path for this cheat-sheet in https://github.com/github/training-kit/pull/427/commits/f2d43853b9f5effd3f96d7694f421293a54deee5 and it should be OK now. I will open a new PR to fix the same issue in the other files.

**Comment by hectorsector** [comment]
Haha, thanks @crichID :man_facepalming:

**Comment by crichID** [comment]
@gh0st I think this looks awesome! Was there anything else you wanted to add before we :ship: it?

**Comment by gh0st** [comment]
@crichID Is there a way I can see it beforehand?

**Comment by brianamarie** [comment]
Hi @gh0st! 

To view this, you'd need to update your branch with the latest changes from training-kit/master, then run `script/server` locally. 

If you're not sure how to update your branch, here are the steps:
1. Add `training-kit` as another remote: `git remote add upstream https://github.com/github/training-kit`
1. Fetch the branches from upstream: `git fetch upstream`
1. While checked out to the branch with changes (which I believe is master) merge in the changes from upstream/master: `git merge upstream/master`. 

After that, just type `script/server` and (pending dependency status in your local environment) you should be able to view the page at `LOCALURL/on-demand/downloads/github-bash-cheat-sheet/`,  but with LOCALURL changing. For me it looks like this: `http://127.0.0.1:4000/on-demand/downloads/github-bash-cheat-sheet/` but for others it's simply `http://4000/on-demand/downloads/github-bash-cheat-sheet/`.

Let me know if you have any questions. It'd be awesome to get this merged in soon, because it's a great cheat sheet! Thank you for all of your work on this!

**Comment by gh0st** [comment]
@brianamarie I am unfortunately still having problems with ruby when I `script/server` so don't wait on me to merge and publish. Thank you for the compliments! You're welcome! I hope the community will see it and make improvements or additions!

**Comment by brianamarie** [comment]
Woohoo! 🎉

**Comment by crichID** [comment]
Hooray! :tada: this is exciting! Thank you for creating this awesome resource @gh0st

**Comment by brianamarie** [comment]
@gh0st You can see the finished product [here](https://services.github.com/on-demand/downloads/github-bash-cheat-sheet/). We still have some work to do to make this resource more discoverable, and hopefully that will be taking place soon!

**Comment by gh0st** [comment]
Very cool! Thank you guys!
